src/database/memory.rs

```diff
@@ -126,6 +128,45 @@ impl MemoryDatabase {
             deleted_keys: Vec::new(),
         }
     }
+
+    fn select_spent_utxos(&self) -> Vec<LocalUtxo> {
+        let key = MapKey::Utxo(None).as_map_key();
+        self.map
+            .range::<Vec<u8>, _>((Included(&key), Excluded(&after(&key))))
+            .filter_map(|(k, v)| {
+                let outpoint = deserialize(&k[1..]).unwrap();
+                let (txout, keychain, is_spent) = v.downcast_ref().cloned().unwrap();
+                if is_spent {
+                    Some(LocalUtxo {
+                        outpoint,
+                        txout,
+                        keychain,
+                        is_spent,
+                    })
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn delete_spent_utxo_set(
+        &mut self,
+        deleted_utxos: &mut Vec<LocalUtxo>,
+        to_del: &Vec<LocalUtxo>,
+    ) -> Result<(), Error> {
```

Comment:
Instead of passing in a mutable list to collect deleted utxos why not just return them in the function call?? Thats the final API anyway. And that will allow you to have your downstream calls more ergonomic.